lib/analysis_server.coffee

```diff
@@ -108,6 +81,7 @@ class AnalysisServer
       callback(obj.event, obj)
 
   setAnalysisRoots: (paths) =>
+    @currentAnalysisRoots = new Set(paths)
```

Comment:
I did some unofficial performance tests, and it's actually pretty snappy. Not sure if it's just me (8GB 2011 MBP) or something else.